extension/public/reddit.js

```diff
@@ -5,95 +5,117 @@ var Reddit = {
+      //image varible contain image url then contain result of ocr
+      let image = this.getDOMImageLink(post);
+
+      if(image != "n/a"){
+          image = await this.ocr(image);
+          image = image.toLowerCase();
+          //callback to wait ocr to be processed
+          callBack();
+      }
+
+      console.log(image);
+      //search if blockedwords are in the title
+      //true if found
+      for (i in blockedWordsData){
+          console.log("pass");
+          if(title.search(blockedWordsData[i]) != -1){
+            return true;
+          }else if (image.search(blockedWordsData[i]) != -1){
+            return true;
+          }else if (description.search(blockedWordsData[i]) != -1){
+            return true;
+          }
+      }
+      return false;
   },
 
-  setupPostUI: function () {
-    // adds hide button to each Reddit post
-    posts = document.querySelectorAll(this.linkSelector);
-    console.log(posts);
-    posts.forEach((post) => {
-      post.innerHTML += this.hideLinkHTML();
-      post
-        .querySelector(".reddit-action.hide")
-        .addEventListener("click", (event) => {
-          this.removeLink(event);
-        });
-    });
-    console.log("setup post UI");
-  },
+  //extract text from image using tesseract
+  //returns text
+  async ocr (url) {
+
+      const { createWorker } = Tesseract;
+      const worker = createWorker ({
+      logger: m => console.log(m)
+      });
+
+      await worker.load();
+      await worker.loadLanguage('eng');
+      await worker.initialize('eng');
+      const { data: { text } } = await worker.recognize(url);
+      await worker.terminate();
+      return text;
 
-  // calls removeLinkFromDOM to remove a specific post link from the DOM
-  removeLink: function (event) {
-    var button = event.currentTarget;
-    var post = button.parentNode.parentNode;
-    var link = post.getAttribute("id");
-
-    this.getDOMTitle(post);
-    this.getDOMLink(post);
-    try {
-      this.getDOMImageLink(post);
-    } catch (e) {
-      console.log("Not an image" + e);
-    }
-    // this.blurImage(post);
-    this.removeLinkFromDOM(link);
   },
 
-  // called removeLinkFromDOM to remove all loaded posts on the page
-  // can use this method to call our image processing on all posts automatically
-  removeLinkAll: function () {
-    var posts = document.querySelectorAll(this.linkSelector);
-    // console.log("Found " + posts.length + " Visible posts");
-    // console.log(posts);
-    posts.forEach((post) => {
-      post.style.display = "none";
-      post.querySelector(".reddit-action").style.display = "none";
-
-    });
+  //blurs the image in a post when blocked keyword is found
+  blurImage: function () {
+      var posts = document.querySelectorAll(this.linkSelector);
+
+      posts.forEach((post) => {
+      var link = post.getAttribute("id");
+      //link(id) of promotions in reddit exceed 50 characters
+      const linkMaxLength = 50;
```

Comment:
Why is this link length limited needed?

Reply:
promotion post spits out an error since these posts are linked to other websites.